test/bench/bench.go

```diff
@@ -40,6 +40,20 @@ type Case struct {
 	Out ref.Val
 }
 
+type EvalCase struct {
```

Comment:
Perhaps call this an `DynamicEnvCase` since this is related to tests related to changes in environment? Also, please add a top-level comment.